### PR TITLE
Use left/right arrow keys to move cursor in IOUAmountPage

### PR DESCRIPTION
**Pull request by rushatgabhane**
cc @roryabraham 
### Details

Replace `onKeyPress` prop with `onChangeText` prop and handle the amount changed.

Also, rename `updateAmountIfValidInput()` to `updateAmountNumberPad()`, and Remove check for `Backspace`.
Because it updates amount only for numpad now. 

### Fixed Issues

$ #3779 

### Tests / QA

1. Request money to any user.
2. Enter an amount.
3. Verify that you can use left and right arrow keys to move the cursor and edit the amount.

### Tested On
Don't have mac to test desktop app.
- [x] Web
- [x] Android
- [x] Mobile Web
- [x] Desktop
- [x] iOS

#### Web

https://user-images.githubusercontent.com/29673073/123718463-d921fd00-d887-11eb-80aa-4faa90d47b5d.mp4

#### Desktop
image

#### iOS
image

### CODE REVIEW

**Comment by roryabraham** [comment]
> Don't have mac to test desktop app.

Appreciate you calling this out! I can test on desktop 🙂

**Comment by roryabraham** [comment]
@rushatgabhane Can you also be sure to test and include screenshots for mobile web and android? I will test on iOS and desktop for you 😄

**Comment by rushatgabhane** [comment]
> @rushatgabhane Can you also be sure to test and include screenshots for mobile web and android? I will test on iOS and desktop for you 😄

Thanks, really appreciate it!
Yup, will add screenshots asap.

File: src/pages/iou/steps/IOUAmountPage.js [hunk 1]
```diff
@@ -102,14 +103,14 @@ class IOUAmountPage extends React.Component {
...
      * @param {String} key
      */
-    updateAmountIfValidInput(key) {
+    updateAmountNumberPad(key) {
         // Backspace button is pressed
-        if (key === '<' || key === 'Backspace') {
+        if (key === '<') {
             if (this.state.amount.length > 0) {
                 this.setState(prevState => ({
                     amount: prevState.amount.substring(0, prevState.amount.length - 1),
```
Comment (roryabraham) [inline src/pages/iou/steps/IOUAmountPage.js:116]: I think this is slightly cleaner (and might fit in one line):

```suggestion
                    amount: prevState.amount.slice(0, -1),
```

File: src/pages/iou/steps/IOUAmountPage.js [hunk 2]
```diff
@@ -132,6 +133,21 @@ class IOUAmountPage extends React.Component {
...
     }
 
+    /**
+     * Update amount on amount change
+     * Validate new amount with decimal number regex up to 6 digits and 2 decimal digit
+     *
+     * @param {String} amount
+     */
+    updateAmount(amount) {
+        // Regex to validate decimal number with up to 3 decimal numbers
```
Comment (roryabraham) [inline src/pages/iou/steps/IOUAmountPage.js:143]: I think we can also DRY this up a bit by creating another function called `validateAmount`

```javascript
validateAmount(amount) {
    const decimalNumberRegex = new RegExp(/^\d+(\.\d{0,3})?$/, 'i');
    return amount === '' || new RegExp(/^\d+(\.\d{0,3})?$/, 'i').test(amount);
}

updateAmountNumberPad(key) {
    // Backspace button is pressed
    if (key === '<') {
        if (this.state.amount.length > 0) {
            this.setState(prevState => ({
                amount: prevState.amount.slice(0, -1),
            }));
        }
        return;
    }
    this.setState((prevState) => {
        // (review note) This might as well be called `amount` instead of `newValue`
        const amount = `${prevState.amount}${key}`;
        return this.validateAmount(amount) ? {amount} : prevState;
    });
}

updateAmount(amount) {
    if (this.validateAmount(amount)) {
        this.setState({amount});
    }
}
```

File: src/pages/iou/steps/IOUAmountPage.js [hunk 3]
```diff
@@ -102,14 +103,14 @@ class IOUAmountPage extends React.Component {
...
+     * Update amount with number or Backspace pressed for BigNumberPad.
      * Validate new amount with decimal number regex up to 6 digits and 2 decimal digit to enable Next button
      *
      * @param {String} key
      */
-    updateAmountIfValidInput(key) {
+    updateAmountNumberPad(key) {
         // Backspace button is pressed
-        if (key === '<' || key === 'Backspace') {
+        if (key === '<') {
```
Comment (roryabraham) [inline src/pages/iou/steps/IOUAmountPage.js:113]: Hmmm why this change?

**Review by roryabraham — CHANGES_REQUESTED** [review]

File: src/pages/iou/steps/IOUAmountPage.js (hunk 3 above)
Comment (rushatgabhane) [reply src/pages/iou/steps/IOUAmountPage.js:113]: Because the `key` can never be `Backspace`.

File: src/pages/iou/steps/IOUAmountPage.js (hunk 2 above)
Comment (rushatgabhane) [reply src/pages/iou/steps/IOUAmountPage.js:143]: Good catch!

Comment (rushatgabhane) [reply src/pages/iou/steps/IOUAmountPage.js]: Tested android, web, mWeb again.

File: src/pages/iou/steps/IOUAmountPage.js [hunk 4]
```diff
@@ -102,6 +102,17 @@ class IOUAmountPage extends React.Component {
         this.unsubscribe();
     }
 
+    /**
+     * Check if amount is a decimal upto 3 digits
+     *
+     * @param {String} amount
+     * @returns {boolean} Return `true` if amount is a decimal upto 3 digits, else `false`
```
Comment (roryabraham) [inline src/pages/iou/steps/IOUAmountPage.js:109]:
```suggestion
     * @returns {Boolean}
```

File: src/pages/iou/steps/IOUAmountPage.js (hunk 3 above)
Comment (roryabraham) [reply src/pages/iou/steps/IOUAmountPage.js:124]: Hmmm but technically you can have a small screen on a device with a keyboard (i.e: make my desktop or browser narrow), and there's no reason specifically to disable this.

**Review by roryabraham — CHANGES_REQUESTED** [review]

File: src/pages/iou/steps/IOUAmountPage.js (hunk 3 above)
Comment (rushatgabhane) [reply src/pages/iou/steps/IOUAmountPage.js:124]: i agree

**Review by roryabraham — APPROVED** [review]
Thanks for making those changes! Great work!

**Comment by roryabraham** [comment]
Unfortunately the deploy comments are not working right now. This was deployed to staging yesterday.
